geode-core/src/main/java/org/apache/geode/management/internal/configuration/functions/GetClusterConfigurationFunction.java

```diff
@@ -32,20 +30,35 @@
 
   @Override
   public void execute(FunctionContext context) {
-    InternalClusterConfigurationService clusterConfigurationService =
-        InternalLocator.getLocator().getSharedConfiguration();
-
     Set<String> groups = (Set<String>) context.getArguments();
+    InternalLocator internalLocator = InternalLocator.getLocator();
+    logger.info("Received request for configuration: {}", groups);
 
-    logger.info("Received request for configuration  : {}", groups);
+    // Return exception to the caller so startup fails fast.
+    if (!internalLocator.isSharedConfigurationEnabled()) {
+      String errorMessage = "The cluster configuration service is not enabled on this member.";
+      logger.warn(errorMessage);
+      context.getResultSender().lastResult(new IllegalStateException(errorMessage));
```

Comment:
A matter of personal preference, but I think this would read better if this block explicitly `return`ed at the end instead of having a long `else` block following it.  The `try` block ends up being five tabs in and that just feels a little heavily nested for my taste.

Reply:
I hear you :-), the change is done.